### DISCUSSION TITLE
USACE Requirements for OpenDCS in CWBI:  Running Legacy Web Pull Programs in CWBI

### DISCUSSION THREAD

**Discussion by jbatterman** [opendcs/opendcs#337] [General]
The Missouri River Basin Water Management Office has brainstormed some requirements for what is needed for USACE to operate OpenDCS in the Cloud.  We are looking for feedback on these requirements, and additions from other offices who have different data acquisition processes.  There will be a new Discussion for each requirement.

Until OpenDCS enhancements to the abstract web pulls are completed, offices need to be able to run their current legacy web pull programs in CWBI.  Some websites are too complex or don’t have a web service associated with them (i.e., NWO’s CNPPID has no web service, NWDM’s METAR data parsing method is too complex to write into a config format statement).

@MikeNeilson @hoffmanjs @opendcs/opendcs-core-devs @opendcs/opendcs-governance @opendcs/opendcs-triage

**Comment by MikeNeilson** [comment]
Do we already have issues in for these?

**Reply by jbatterman** [reply]
I haven't opened issues for these but will do so today.  @hoffmanjs if I miss any please let me know or add them as well.   These will just be for actual data pulls, not other programs that grab other types of data (like rating curves), unless you envision those processes also running through OpenDCS.

**Reply by MikeNeilson** [reply]
That's worth investigating but since we (USACE) have things that work it'll likely be lower on the priority list unless other users ask for it also.

**Reply by jbatterman** [reply]
See following issues.  If there is a way to continue running these data ingest programs on a schedule in CWBI we would also consider that alternative.  As a side note, there may be other ingest processes that aren't included here but can't be parsed through OpenDCS right now.  We've ported 6 different web sources to OpenDCS but that still doesn't include all the processes that I believe can be parsed through OpenDCS with its current functionality.  If I'm wrong there, there will be more sources that need additional features to be parsed through OpenDCS.

https://github.com/opendcs/opendcs/issues/338
https://github.com/opendcs/opendcs/issues/339

**Reply by MikeNeilson** [reply]
METAR probably makes the most sense as it's something well defined, HTML scraping could be a possiblity, but something in python running in Apache Airflow (currently the proposed cron replacement on CWBI) would likely be easier to maintain.
